src/arrayMethodSort.js

```diff
@@ -6,6 +6,30 @@
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
     // write code here
+    const array = this;
```

Comment:
Directly reassigning 'this' to another variable is not a good practice. Use `this` directly in this task